📝 add rum-react documentation to the integration page (#2604)
DataDog/integrations-extras: rum_react/README.md

## Overview

With the Datadog RUM React integration, resolve performance issues quickly in React components by:

- Debugging the root cause of performance bottlenecks, such as a slow server response time, render-blocking resource, or an error inside a component

> - Automatically correlating Web performance data with user journeys, HTTP calls, and logs

Reviewer: ```suggestion
- Automatically correlating web performance data with user journeys, HTTP calls, and logs
```

> - Alerting your engineering teams when crucial Web performance metrics (such as Core Web Vitals) fall below a threshold that results in a poor user experience

Reviewer: ```suggestion
- Alerting your engineering teams when crucial web performance metrics (such as Core Web Vitals) fall below a threshold that results in a poor user experience
```

Monitor your React applications from end-to-end by:

- Tracking and visualizing user journeys across your entire stack
- Debugging the root cause of slow load times, which may be an issue with your React code, network performance, or underlying infrastructure
- Analyzing and contextualizing every user session with attributes such as user ID, email, name, and more
- Unifying full-stack monitoring in one platform for frontend and backend development teams

## Setup

> Make sure to [setup Datadog RUM][1] in your application. When creating a new RUM application within the Datadog App, make sure to select React. You can also edit an existing RUM application and change its type to React. The Datadog App then shows you the instructions to setup the Browser SDK [RUM-React plugin][2].

Reviewer 2: We could have this section similar to the Browser one: https://docs.datadoghq.com/real_user_monitoring/browser/setup/client/?tab=rum

To set up React Monitoring, create an application in Datadog:

1. In Datadog, navigate to the [**Digital Experience** > **Add an Application**][1] page and select the React application type.
   - By default, automatic user data collection is enabled. To disable automatic user data collection for either client IP or geolocation data, uncheck the boxes for those settings. For more information, see [RUM Browser Data Collected][2].
   - Enter a name for your application and click **Generate Client Token**. This generates a `clientToken` and an `applicationId` for your application.
   - Define the environment name and service name for your application to use unified service tagging for [RUM & Session Replay][3]. Set a version number for your deployed application in the initialization snippet. For more information, see [Tagging](#tagging).
   - Set the sampling rate of total user sessions collected and use the slider to set the percentage of total [Browser RUM & Session Replay][4] sessions collected. Browser RUM & Session Replay sessions include resources, long tasks, and replay recordings. For more information about configuring the percentage of Browser RUM & Session Replay sessions collected from the total amount of user sessions, see [Configure Your Setup For Browser and Browser RUM & Session Replay Sampling][4].
   - Click the **Session Replay Enabled** toggle to access replay recordings in [Session Replay][5].
   - Select a [privacy setting][6] for Session Replay in the dropdown menu.
2. Deploy the changes to your application. Once your deployment is live, Datadog collects events from your users' browsers.
3. Visualize the [data collected][3] in [dashboards][7] or create a search query in the [RUM Explorer][8].
4. (Optional) Initialize the RUM SDK with the `allowedTracingUrls` parameter to [Connect RUM and Traces][9] if you want to start linking requests from your web and mobile applications to their corresponding backend traces. See the full list of [initialization parameters](#initialization-parameters).
5. If you're using the Datadog Content Security Policy (CSP) integration on your site, see [the RUM section of the CSP documentation][10] for additional setup steps.

Until Datadog starts receiving data, your application appears as `pending` on the **RUM Applications** page.

Author: The link on "setup Datadog RUM" is already pointing to that doc. I don't think we should duplicate it here.

Reviewer 2: Okay, I didn’t see that. Maybe we should rephrase a bit to not have so many "make sure"

We should structure it more clearly to avoid confusion—kind of like those tricky instructions:

"In front, you have the water cap. (You open it.) Make sure you do, don’t open it. (You close it.) If you close it, make sure you don’t close it." 😆

-- 

Start by setting up [Datadog RUM][1] in your React application. If you're creating a new RUM application in the Datadog App, select React as the application type. If you already have an existing RUM application, you can update its type to React instead. Once configured, the Datadog App will provide instructions for integrating the [RUM-React plugin][2] with the Browser SDK.

Author: Thanks, I agree your version is clearer. I commited it.

> Make sure to [setup Datadog RUM][1] in your application. When creating a new RUM application within the Datadog App, make sure to select React. You can also edit an existing RUM application and change its type to React. The Datadog App will then show you the instructions to setup the Browser SDK [RUM-React plugin][2].

Reviewer: ```suggestion
Make sure to [setup Datadog RUM][1] in your application. When creating a new RUM application within the Datadog App, make sure to select React. You can also edit an existing RUM application and change its type to React. The Datadog App then shows you the instructions to setup the Browser SDK [RUM-React plugin][2].
```

## Error Tracking

To track React component rendering errors, use one of the following:

- An `ErrorBoundary` component (see [React documentation][3]) that catches errors and reports them to Datadog.
- A function that you can use to report errors from your own `ErrorBoundary` component.

#### `ErrorBoundary` usage

```javascript
import { ErrorBoundary } from '@datadog/browser-rum-react'

function App() {
  return (
    <ErrorBoundary fallback={ErrorFallback}>
      <MyComponent />
    </ErrorBoundary>
  )
}

> function ErrorFallback({ resetError, error }: { resetError: () => void; error: unknown }) {

Reviewer 2: Maybe is better to remove the types if is not adding any information.

Author: Thanks for the suggestion, I removed the types

## Overview

With the Datadog RUM React integration, resolve performance issues quickly in React components by:

- Debugging the root cause of performance bottlenecks, such as a slow server response time, render-blocking resource, or an error inside a component
- Automatically correlating Web performance data with user journeys, HTTP calls, and logs
- Alerting your engineering teams when crucial Web performance metrics (such as Core Web Vitals) fall below a threshold that results in a poor user experience

Monitor your React applications from end-to-end by:

- Tracking and visualizing user journeys across your entire stack
- Debugging the root cause of slow load times, which may be an issue with your React code, network performance, or underlying infrastructure
- Analyzing and contextualizing every user session with attributes such as user ID, email, name, and more
- Unifying full-stack monitoring in one platform for frontend and backend development teams

## Setup

First, make sure to [setup Datadog RUM][1] in your application.

> Then, install the `@datadog/browser-rum-react` NPM package using your package manager of choice. For example:

Reviewer 3: Should we point to the RUM Application creation page here instead of repeating installation steps? That would avoid having to maintain the instructions in both places.

Author: Interesting. I changed the setup section, let me know what you think!
